nats/nats.go

```diff
@@ -56,16 +58,28 @@ func newConn(stanClusterID, clientID string, opts []stan.Option) (*conn, error)
 	}, nil
 }
 
-// Topic returns producer topic.
 func (c *conn) Topic(name string) bps.Topic {
 	return &topic{
 		stan: c.stan,
 		name: name,
 	}
 }
 
-// Subscribe subscribes to topic messages.
-func (c *conn) Subscribe(ctx context.Context, topic string, handler bps.Handler) error {
+func (c *conn) Subscribe(ctx context.Context, topic string, handler bps.Handler, options ...bps.SubOption) error {
+	opts := (&bps.SubOptions{
+		StartAt: bps.PositionNewest,
+	}).Apply(options)
+
+	var startPos pb.StartPosition
+	switch opts.StartAt {
+	case bps.PositionNewest:
+		startPos = pb.StartPosition_NewOnly
+	case bps.PositionOldest:
+		startPos = pb.StartPosition_First
+	default:
+		return fmt.Errorf("start position %d is not supported by this implementation", opts.StartAt)
```

Comment:
suggestion: use `%s`, not `%d`, either add a `String()` function to `bps.StartPosition` or make the enum values themselves strings?

Reply:
Will change StartPosition to string.